File: app/src/main/java/rod/bailey/trafficatnsw/cameras/filter/AdmitRegionalTrafficCameraFilter.java
package rod.bailey.trafficatnsw.cameras.filter;

import rod.bailey.trafficatnsw.cameras.TrafficCamera;

public class AdmitRegionalTrafficCameraFilter implements ITrafficCameraFilter {

	@Override
	public boolean admit(TrafficCamera camera) {
		return (camera.getRegion().isRegional());
	}

}

File: app/src/main/java/rod/bailey/trafficatnsw/cameras/filter/AdmitRegionalTrafficCameraFilter.kt
package rod.bailey.trafficatnsw.cameras.filter

import rod.bailey.trafficatnsw.cameras.data.XCamera
import rod.bailey.trafficatnsw.hazard.data.XRegion

class AdmitRegionalTrafficCameraFilter : ITrafficCameraFilter {
	override fun admit(camera: XCamera): Boolean {
		val regionStr: String? = camera.properties?.region
		if (regionStr != null) {
			val xregion: XRegion = XRegion.valueOf(regionStr)
			return xregion.isRegional
		}
		return false
	}
}
